lib/packwerk/cli/command_registry.rb

```diff
@@ -0,0 +1,64 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  class Cli
+    class CommandRegistry
+      extend T::Sig
+
+      class << self
+        extend T::Sig
+
+        sig { params(name: String, help: String, aliases: T::Array[String]).void }
+        def register(name, help:, aliases: [])
+          registry << new(name, help: help, aliases: aliases)
+        end
+
+        sig { params(name_or_alias: String).returns(T.nilable(T.class_of(Cli::BaseCommand))) }
+        def class_for(name_or_alias)
+          registry
+            .find { |command| command.matches_command?(name_or_alias) }
+            &.command_class
+        end
+
+        sig { returns(T::Array[CommandRegistry]) }
+        def all
+          registry.dup
+        end
+
+        private
+
+        sig { returns(T::Array[CommandRegistry]) }
+        def registry
+          @registry ||= T.let([], T.nilable(T::Array[CommandRegistry]))
+        end
+      end
+
+      sig { returns(String) }
+      attr_reader :name
+
+      sig { returns(String) }
+      attr_reader :help
+
+      sig { params(name: String, help: String, aliases: T::Array[String]).void }
+      def initialize(name, help:, aliases: [])
+        @name = name
+        @help = help
+        @aliases = aliases
+      end
+
+      sig { returns(T.class_of(Cli::BaseCommand)) }
+      def command_class
+        classname = @name.sub(" ", "_").underscore.classify + "Command"
+        Cli.const_get(classname) # rubocop:disable Sorbet/ConstantsFromStrings
```

Comment:
Can you explain your motivation to not use a `Packwerk::Cli::Commands` module to const_get from? We could fit all command related code in one place (eg. `Commands::Registry`, `Commands::Base`, `Commands::HelpCommand`, etc. It seems safer and more organized.

Reply:
Without the need to auto match a command name with a class name, and with instead an explicit registration of each command, I felt `Commands` had lost its main purpose, that triggered its creation: hosting all commands and nothing else so that `packwerk not-a-command-but-const-with-matching-name-exists` would fail beautifully and not weirdly.
With its main purpose lost, I went for the "less code is better" approach.

Reply:
Personally, I think the organization of the `commands/` namespace makes command specific code easier to reason about, but I don't feel strongly about it. I'm slightly worried about this from a public API perspective.

Reply:
`Cli` is public, sure, but if I make my constants nested under it private, it shouldn't make a difference?
I'm not antagonizing, just trying to understand where the problem would be exactly.

Reply:
https://github.com/Shopify/packwerk/commit/670fc67b7413c5a03cc6896193ca64754228c9bd